IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.m

```diff
@@ -446,32 +446,34 @@ - (void)decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction
     if (self.customHeaderProvider)
     {
         [self.customHeaderProvider getCustomHeaders:navigationAction.request
-                                                    forHost:requestURL.host
-                                            completionBlock:^(NSDictionary<NSString *, NSString *> *extraHeaders, NSError *error){
-            if (extraHeaders && extraHeaders.count > 0)
-            {
-                NSMutableURLRequest *newUrlRequest = [navigationAction.request mutableCopy];
-                
-                for (NSString *headerKey in extraHeaders)
+                                            forHost:requestURL.host
+                                    completionBlock:^(NSDictionary<NSString *, NSString *> *extraHeaders, NSError *error){
+            [MSIDMainThreadUtil executeOnMainThreadIfNeeded:^{
```

Comment:
The only change in this PR is adding line 451 `[MSIDMainThreadUtil executeOnMainThreadIfNeeded:^{`.

The whole `decidePolicyForNavigationAction` delegate is executed in main thread. However, after we come back from line 450, it is not guranteed to be in main thread anymore.